src/lib/sql/sql_pipeline_statement.hpp

```diff
@@ -62,9 +62,15 @@ class SQLPipelineStatement : public Noncopyable {
   // This can be a nullptr if no transaction management is wanted.
   const std::shared_ptr<TransactionContext>& transaction_context() const;
 
+  // Return the duration of each step in the pipeline.
+  std::chrono::microseconds translate_time_microseconds() const;
```

Comment:
We don't have to do this in this PR, but wouldn't it be cleaner to return a struct that holds all the information like the runtime of the different steps, whether the cache was hit, etc.?

Reply:
probably yes. I'll do that in a separate PR.